packages/block-editor/src/components/editor-styles/index.js

```diff
@@ -1,62 +1,52 @@
 /**
  * External dependencies
  */
-import { compact, map } from 'lodash';
 import tinycolor from 'tinycolor2';
 
 /**
  * WordPress dependencies
  */
-import { useCallback, useRef } from '@wordpress/element';
+import { useCallback } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import transformStyles from '../../utils/transform-styles';
 
-function syncDarkThemeBodyClassname( node ) {
-	const backgroundColor = window
-		.getComputedStyle( node, null )
-		.getPropertyValue( 'background-color' );
-
-	const { ownerDocument } = node;
-	const body = ownerDocument.getElementsByTagName( 'body' )[ 0 ];
-
-	if ( tinycolor( backgroundColor ).getLuminance() > 0.5 ) {
-		body.classList.remove( 'is-dark-theme' );
-	} else {
-		body.classList.add( 'is-dark-theme' );
-	}
+const EDITOR_STYLES_CLASS_NAME = '.editor-styles-wrapper';
+
+function useDarkThemeBodyClassName( dependencies ) {
+	return useCallback( ( node ) => {
+		if ( ! node ) {
+			return;
+		}
+
+		const { ownerDocument } = node;
+		const { defaultView, body } = ownerDocument;
+		const canvas = ownerDocument.querySelector( EDITOR_STYLES_CLASS_NAME );
+		const backgroundColor = defaultView
+			.getComputedStyle( canvas, null )
+			.getPropertyValue( 'background-color' );
+
+		if ( tinycolor( backgroundColor ).getLuminance() > 0.5 ) {
+			body.classList.remove( 'is-dark-theme' );
+		} else {
+			body.classList.add( 'is-dark-theme' );
+		}
+	}, dependencies );
 }
 
-export default function useEditorStyles( styles ) {
-	const nodes = useRef( [] );
-
-	return useCallback(
-		( node ) => {
-			if ( ! node ) {
-				nodes.current.forEach( ( styleElement ) =>
-					styleElement.ownerDocument.body.removeChild( styleElement )
-				);
-				return;
-			}
-
-			const updatedStyles = transformStyles(
-				styles,
-				'.editor-styles-wrapper'
-			);
-
-			const { ownerDocument } = node;
-			nodes.current = map( compact( updatedStyles ), ( updatedCSS ) => {
-				const styleElement = ownerDocument.createElement( 'style' );
-				styleElement.innerHTML = updatedCSS;
-				ownerDocument.body.appendChild( styleElement );
-
-				return styleElement;
-			} );
-
-			syncDarkThemeBodyClassname( node );
-		},
-		[ styles ]
+export default function EditorStyles( { styles } ) {
+	return (
+		<>
+			{ /* Use an empty style element to have a document reference,
+			     but this could be any element. */ }
+			<style ref={ useDarkThemeBodyClassName( styles ) } />
+			{ transformStyles( styles, EDITOR_STYLES_CLASS_NAME ).map(
```

Comment:
The only behavioral change I've noticed here is that we no longer remove falsy values from the array. I'm unsure what use case this aimed to cover, given the styles are provided by `add_editor_styles`. Flagging in case you know more about this.